compiler/rustc_typeck/src/check/pat.rs

```diff
@@ -1421,7 +1421,22 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     plural
                 ),
             );
-            if plural == "" {
+
+            if inexistent_fields.len() == 1 && unmentioned_fields.len() == 1 {
+                let unmentioned_field = unmentioned_fields[0].1.name;
+                err.span_suggestion(
+                    ident.span,
+                    &format!(
+                        "`{}` has a field named `{}`, try replacing `{}` for it.",
+                        tcx.def_path_str(variant.def_id),
+                        unmentioned_field,
+                        inexistent_fields[0]
+                    ),
+                    format!("{}", unmentioned_field),
```

Comment:
This can be simplified to
```suggestion
                    unmentioned_field.to_string(),
```